File: example/java-lambda/src/main/java/com/lamblin/example/lambda/LambdaHandler.java
package com.lamblin.example.lambda;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
import com.google.common.collect.ImmutableSet;
import com.lamblin.core.Lamblin;
import com.lamblin.example.lambda.controller.TodoController;
import com.lamblin.example.lambda.service.DataService;
import com.lamblin.example.lambda.service.InMemoryDataService;

import java.io.InputStream;
import java.io.OutputStream;

public class LambdaHandler implements RequestStreamHandler {

  private static final DataService DATA_SERVICE = new InMemoryDataService();
  private static final TodoController TODO_CONTROLLER = new TodoController(DATA_SERVICE);

  public void handleRequest(InputStream input, OutputStream output, Context context) {
    Lamblin lamblin = Lamblin.instance(ImmutableSet.of(TODO_CONTROLLER));

    lamblin.handlerRequest(input, output);
  }
}

File: example/kotlin-lambda/src/main/kotlin/com/lamblin/example/lambda/LambdaHandler.kt
package com.lamblin.example.lambda

import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.RequestStreamHandler
import com.lamblin.core.Lamblin
import com.lamblin.example.lambda.controller.TodoController
import com.lamblin.example.lambda.service.InMemoryDataService
import java.io.InputStream
import java.io.OutputStream

val todoController = TodoController(InMemoryDataService)

/** Defines the main Handler class used by Lambda. */
class LambdaHandler : RequestStreamHandler {

    override fun handleRequest(input: InputStream, output: OutputStream, p2: Context) {
        val lamblin = Lamblin.instance(setOf(todoController))

        lamblin.handlerRequest(input, output)
    }

}